pkg/inspector/image-inspector.go

```diff
@@ -166,27 +166,40 @@ func aggregateBytesAndReport(bytesChan chan int) {
 // decodeDockerPullMessages will parse the docker pull messages received
 // from reader and will push the difference of bytes downloaded to
 // bytesChan. After reader is closed it will close bytesChan and exit.
-func decodeDockerPullMessages(bytesChan chan int, reader io.Reader) {
+func decodeDockerPullMessages(parsedErrors chan bool, bytesChan chan int, reader io.Reader) {
 	type progressDetailType struct {
 		Current, Total int
 	}
 	type pullMessage struct {
 		Status, Id     string
 		ProgressDetail progressDetailType
+		Error          string
 	}
 	defer func() { close(bytesChan) }()           // Closing the channel to end the other routine
+	defer func() { close(parsedErrors) }()        // Close channel
```

Comment:
closing the channels here rather than allowing the creator to do it is suspicious